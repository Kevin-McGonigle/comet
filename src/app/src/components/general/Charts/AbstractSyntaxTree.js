import React from 'react';
import TreeGraph from '../TreeGraph/TreeGraph';
import { Pane } from 'evergreen-ui';

const AbstractSyntaxTree = props => {
    const { data } = props;
    
    return (
        <Pane
            display="flex"
            flexWrap="wrap"
            flexGrow={1}
            marginRight="10"
            background="tint2"
            elevation={4}   
        >
           <TreeGraph tree={data}/>
        </Pane>
    )
};

export default AbstractSyntaxTree;
