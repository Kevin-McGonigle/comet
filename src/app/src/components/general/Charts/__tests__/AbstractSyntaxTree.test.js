import React from 'react';
import AbstractSyntaxTree from '../AbstractSyntaxTree';
import {shallow} from 'enzyme';
import "../../../../setupTests"

const AST = {
    name: "Statements",
    children: [
        {
            name: "if statement",
            children: [{
                name: "true"
            }, {
                name: "Pass statement",
                children: null
            }]
        }
    ]
}

test('should render ForceDirectedGraph as expected', () => {
    const component = shallow(
        <AbstractSyntaxTree
            data={AST}
        />
    );
    expect(component.html()).toMatchSnapshot();
});
