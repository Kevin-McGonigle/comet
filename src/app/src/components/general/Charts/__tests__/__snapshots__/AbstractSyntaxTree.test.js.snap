// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ForceDirectedGraph as expected 1`] = `"<div class=\\"css-nil ðŸ“¦bs_kymczs ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-wrap_wrap ðŸ“¦flx-grow_1 ðŸ“¦mr_10 ðŸ“¦box-szg_border-box\\"><div class=\\"treeGraphContainer\\"><svg transform=\\"rotate(90)\\" height=\\"900\\" width=\\"800\\"><path class=\\"link\\" d=\\"M10,440C167.5,440 167.5,440 325,440\\"></path><path class=\\"link\\" d=\\"M325,440C482.5,440 482.5,220 640,220\\"></path><path class=\\"link\\" d=\\"M325,440C482.5,440 482.5,660 640,660\\"></path><g class=\\"node\\" transform=\\"translate(10, 440)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">Statements</text></g><g class=\\"node\\" transform=\\"translate(325, 440)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">if statement</text></g><g class=\\"node\\" transform=\\"translate(640, 220)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">true</text></g><g class=\\"node\\" transform=\\"translate(640, 660)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">Pass statement</text></g></svg></div></div>"`;
